### PR TITLE
[pruner config] make repeated parameters optional/default + add metric

### PR DESCRIPTION
**Pull request by phoenix-o**
* makes repeatable parameters optional (num_latest_epoch_dbs_to_retain, epoch_db_pruning_period_secs, max_checkpoints_in_batch, max_transactions_in_batch)
* adds metrics to represent following config parameters: num_epochs_to_retain, num_epochs_to_retain_for_checkpoints

### CODE REVIEW

File: crates/sui-core/src/authority/authority_store_pruner.rs [hunk 1]
```diff
@@ -375,6 +389,8 @@ impl AuthorityStorePruner {
         let mut checkpoints_to_prune = vec![];
         let mut checkpoint_content_to_prune = vec![];
         let mut effects_to_prune = vec![];
+        let max_checkpoints_in_batch = config.max_checkpoints_in_batch.unwrap_or(10);
```
Comment (halfprice) [inline crates/sui-core/src/authority/authority_store_pruner.rs:392]: nit: I wonder if it may help with readability if instead of making these config an Option type, we use the serde default function to explicitly set the default value in the config struct, so that future reader can quickly see the default value.

When I first read the code few weeks back, it took me quite sometime to realize that AuthorityStorePruningConfig::Default() is not really the default that will be applied in prod. But rather, the value in each `unwrap_or()` in the code where the config is accessed is.

**Review by halfprice — APPROVED** [review]

**Review by longbowlu — COMMENTED** [review]
can we have the default values as constants in top of the file or a centralized place?

**Comment by longbowlu** [comment]
> makes repeatable parameters optional

what does `repeatable` mean?

**Review by mwtian — APPROVED** [review]
LGTM but as @halfprice mentioned it would be nice to have consistency between AuthorityStorePruningConfig::default() and default deserialized values.

**Comment by mwtian** [comment]
> can we have the default values as constants in top of the file or a centralized place?

Another idea is to put default values in functions for serde default, and use those function to implement `AuthorityStorePruningConfig::default()`.

**Comment by XoL1507** [comment]
thanks for the effort..
